plugins/kubernetes/README.md

```diff
@@ -222,3 +222,11 @@ this is useful when dealing with large deployments that have some random failure
 To debug services or to create resources that needs to reference a selector that doesn't include team/role (like a Gateway), you can disable selector validation with:
 
 `metadata.annotations.samson/service_selector_across_roles: "true"`
+
+### Ordering environment variables based on references
+
+In kubernetes, you can reference an environment variable from another using the syntax: `FOO: $(BAR)`. However, in this case `BAR` needs to be defined before `FOO` for it to work. This is due to an implementation detail in kubernetes itself.
```

Comment:
```suggestion
In kubernetes, you can reference an environment variable from another using the syntax: `FOO: ${BAR}`. However, in this case `BAR` needs to be defined before `FOO` for it to work. This is due to an implementation detail in kubernetes itself.
```

Reply:
... maybe I understood your proposal wrong ... 
samson does env variable replacement too with `${FOO}` which works for non-kubernetes projects too

Reply:
Basically, what we want is to be able to reference `$DEPLOY_GROUP`, which is not available when the `env`-plugin runs it's `resolve_dollar_variables()`-method in `EnvironmentVariable`, triggered by the `Samson::Hook` in the `kubernetes`-plugin.

We decided to make this an optional feature to please kubernetes, enabled by setting a environment variable, rather than refactoring how the `kubernetes` and `env` (and possibly other) plugins pass around env vars through hooks without knowing if that was a appropriate design decision.

What do you think @grosser ?

Reply:
I think the problem is that resolve_dollar_variables (and preview / resolve-secrets) belongs into base and not into the env plugins since we should only resolve once we have collected all env vars from base+plugins.

A solution would be to add a new `Samson::EnvResolver` class that collects env (can be given a base env too) and handles preview/merge/resolve/secrets ... then use that from template_filler and from plugins/env/app/controllers/env/environment_controller.rb.

... with that any other plugin can add env vars via deploy_group_env hook and they automatically get resolved/previewed

... sounds good ?

(base make a fresh PR for that so this here does not get too cluttered)